instances/widgets.treasury-factory.near/widget/pages/payments/ProposalDetailsPage.jsx

```diff
@@ -182,6 +194,267 @@ useEffect(() => {
+    }
+  }
+}, [proposalData?.isIntentsPayment, networkInfo.blockchain]);
+
+// Note: No fallback target network explorer links - only show actual transaction links
+
+// Fetch target chain transaction hash for approved intents payments
+useEffect(() => {
+  if (
+    proposalData?.isIntentsPayment &&
+    proposalData?.status === "Approved" &&
+    networkInfo.blockchain &&
+    networkInfo.blockchain !== "near" &&
+    transactionInfo.nearTxHash &&
+    transactionInfo.nearTxHash.includes("/txns/")
+  ) {
+    // Extract the transaction hash from the URL
+    const nearTxHash = transactionInfo.nearTxHash.split("/txns/")[1];
+
+    console.log(`Fetching withdrawal status for NEAR tx: ${nearTxHash}`);
+
+    // Call POA Bridge API to get withdrawal status
+    asyncFetch("https://bridge.chaindefuser.com/rpc", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: 1,
+        method: "withdrawal_status",
+        params: [
+          {
+            withdrawal_hash: nearTxHash,
+          },
+        ],
+      }),
+    })
+      .then((response) => {
+        if (!response || !response.body) {
+          console.log("No withdrawal status response");
+          return;
+        }
+
+        const result = response.body.result;
+        if (
+          result &&
+          result.status === "COMPLETED" &&
+          result.data?.transfer_tx_hash
+        ) {
+          console.log(`Found target chain tx: ${result.data.transfer_tx_hash}`);
+
+          // Create target chain explorer link
+          let targetExplorerLink = null;
+          if (result.data.chain && result.data.transfer_tx_hash) {
+            const chainInfo = result.data.chain.toLowerCase();
+            if (chainInfo.includes("eth")) {
+              targetExplorerLink = `https://etherscan.io/tx/${result.data.transfer_tx_hash}`;
+            } else if (chainInfo.includes("polygon")) {
+              targetExplorerLink = `https://polygonscan.com/tx/${result.data.transfer_tx_hash}`;
+            } else if (chainInfo.includes("bsc")) {
+              targetExplorerLink = `https://bscscan.com/tx/${result.data.transfer_tx_hash}`;
+            }
+            // Add more chains as needed
+          }
+
+          if (targetExplorerLink) {
+            setTransactionInfo((prev) => ({
+              ...prev,
+              targetTxHash: targetExplorerLink,
+            }));
+          }
+        } else {
+          console.log(`Withdrawal status: ${result?.status || "unknown"}`);
+        }
+      })
+      .catch((error) => {
+        console.error("Error fetching withdrawal status:", error);
+      });
+  }
+}, [
+  proposalData?.isIntentsPayment,
+  proposalData?.status,
+  transactionInfo.nearTxHash,
+  networkInfo.blockchain,
+]);
+
+// Fetch execution transaction hash for approved proposals
+useEffect(() => {
+  if (
+    proposalData?.status === "Approved" &&
```

Comment:
may be we should do it for`'Failed'` as well?

Reply:
Yes absolutely! Thanks for seeing that. Added now

<img width="1196" alt="image" src="https://github.com/user-attachments/assets/437344fb-975e-497a-ae55-ea6bb7ba0837" />